app/models/vehicle.rb

```diff
@@ -4,7 +4,10 @@ class Vehicle < ApplicationRecord
 
   validates :model, length: { minimum: 3 }
   validates :air_conditioning, inclusion: { in: [true, false] }
-  enum vehicle_type: { car: 0, van: 1, boat: 2, yatch: 3, motorbike: 4, bike: 5, caravan: 6 }
   validates :consumption, presence: true, numericality: { only_float: true, greater_than: 0 }
   validates :nickname, presence: true, length: { minimum: 2, maximum: 10 }
+
+  enum vehicle_type: { car: 0, van: 1, boat: 2, yatch: 3, motorbike: 4, bike: 5, caravan: 6 }
+
+  scope :user_vehicles, ->(user_id) { select(:nickname).where("user_id = ?", user_id) }
```

Comment:
I'm afraid this scope has not a test. moreover, there is something not very clear here.. if you want to retrieve the vehicle nickname you don't need a scope... conversely if you need the user nickname you are not getting it.

Reply:
at the end we realized we don¡'t need this 😄